GitUI/UserControls/FileStatusList.cs

```diff
@@ -948,6 +1031,32 @@ string GetDescriptionForRevision(ObjectId objectId)
             }
         }
 
+        private void UpdateColumnWidth()
+        {
+            Task.Run(async () =>
+            {
+                // by changing the width after delay we workaround ListView bug
+                // that renders ListView unusable if column Width is set within any
+                // event handler like SizeChanged, ClientSizeChanged, Layout and etc.
+                // https://github.com/gitextensions/gitextensions/issues/5437
+                await Task.Delay(TimeSpan.FromMilliseconds(10));
+
+                Invoke((Action)delegate
+                {
+                    FileStatusListView.BeginUpdate();
+                    columnHeader.AutoResize(ColumnHeaderAutoResizeStyle.ColumnContent);
+
+                    var minWidth = FileStatusListView.ClientSize.Width;
+                    if (columnHeader.Width < minWidth)
+                    {
+                        columnHeader.Width = minWidth;
+                    }
+
+                    FileStatusListView.EndUpdate();
+                });
+            });
+        }
```

Comment:
@sharwell / @jbialobr is this implementation ok, or should this be written something like:

```
            this.InvokeAsync(() =>
            {
                // by changing the width after delay we workaround ListView bug
                // that renders ListView unusable if column Width is set within any
                // event handler like SizeChanged, ClientSizeChanged, Layout and etc.
                // https://github.com/gitextensions/gitextensions/issues/5437
                Thread.Sleep(TimeSpan.FromMilliseconds(10));

                FileStatusListView.BeginUpdate();
                columnHeader.AutoResize(ColumnHeaderAutoResizeStyle.ColumnContent);

                var minWidth = FileStatusListView.ClientSize.Width;
                if (columnHeader.Width < minWidth)
                {
                    columnHeader.Width = minWidth;
                }

                FileStatusListView.EndUpdate();
            }).FileAndForget();
```

Reply:
Yes, thanks, I had a brain fart after a long day. 🤦‍♂️ 
I was thinking of JTF